tests/test_featureset.py

```diff
@@ -845,3 +846,68 @@ def test_convert_featureset():
                                                           '.csv', '.arff',
                                                           '.libsvm'], 2):
         yield check_convert_featureset, from_suffix, to_suffix
+
+
+def featureset_creation_from_dataframe_helper(with_labels):
+    """
+    Helper function for the two unit tests for FeatureSet.from_data_frame().
+    Since labels are optional, run two tests, one with, one without.
```

Comment:
Take a look at how it is created in the skll ``utils.py`` method ``make_classification_data`` and see whether you could apply that here too on your data?